Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -25,30 +25,68 @@
 
 #include "config.h"
 #include "ScrollAnchoringController.h"
+#include <wtf/text/TextStream.h>
 
 namespace WebCore {
 
-void ScrollAnchoringController::invalidateAnchorElement()
+void ScrollAnchoringController::invalidateAnchorBox()
 {
-    m_anchorElement = nullptr;
-    m_lastPositionForAnchorElement = FloatPoint();
+    m_anchorBox = nullptr;
+    m_lastPositionForAnchorBox = FloatSize();
 }
 
-void ScrollAnchoringController::selectAnchorElement()
+FloatSize ScrollAnchoringController::computeOffset(RenderBox* candidate)
 {
-    // TODO: implement
-    // perform Anchor Element Selection algorithm on dom elements within frame view's viewable region
-    // Firefox appears to only iterate through frame tree under the current frame (unclear if
-    // looking through frame tree is sufficient)
-    // unclear to me where to call this from 
+    return candidate->location() - m_frameView.rectForFixedPositionLayout().location();
+}
+
+void ScrollAnchoringController::selectAnchorBox()
+{
+    // TODO: fully implement anchor node selection algorithm in spec
+    // (specifically excluded subtrees and priority nodes)
+    auto scrollPosition = frameView().scrollPosition();
+    if (m_anchorBox || scrollPosition == IntPoint())
+        return;
+    
+    auto& document = *m_frameView.frame().document();
+    auto* documentElement = document.documentElement();
+    RenderBox* rootRenderer = documentElement ? documentElement->renderBox() : nullptr;
+        RenderBox* candidateAnchor = nullptr;
+    auto viewableRect = m_frameView.rectForFixedPositionLayout();
+    
+    Vector<RenderBox*> tree;
+    tree.append(rootRenderer->firstChildBox());
+    while (!tree.isEmpty()) {
+        auto box = tree.last();
+        tree.removeLast();
+        if (box->style().overflowAnchor() == OverflowAnchor::None)
+            continue;
+        auto boxRect = box->frameRect();
```

Comment:
frameRect is local to the box. You need to convert to `absolute` coordinates if you're comparing with the layoutViewport (I think; you should check this, and test with zooming).

Reply:
Fixed.